### ISSUE TITLE
Python 3 support

### ISSUE THREAD

**Issue by andyDoucette** [dkobozev/tatlin#23]
The last python 2 train has already left the building.  Only python 3 trains from here on out.  Can this get ported to python 3?  Would you accept such pull requests?

**Comment by dkobozev** [comment]
That would be great! I didn't realize anyone was still using Tatlin, tbh :)

**Comment by andyDoucette** [comment]
haha!  I love it. :)  It's just what it should be - a versatile viewer that
makes it easy to see the scale of a part, it's dimensions, and rotate
things.  I have probably referred 200 people over the years.

**Comment by dkobozev** [comment]
wxPython seems to be alive and well, and so does PyOpenGL. Not sure how much has changed in those APIs over the years, but at least it doesn't seem impossible to port the app over to Python 3 without a complete rewrite.

**Comment by Blokkendoos** [comment]
Here is one of those tatlin users. Just like Andy said, a simple versatile viewer.
Anyway, today I created a Python3 translation, see: https://github.com/Blokkendoos/tatlin
Before creating a pull-request, I will use it a while first.

**Comment by andyDoucette** [comment]
Yay!  Thanks so much for that!  I miss using tatlin and would love to
switch back now that I can run it using python3.

**Comment by Blokkendoos** [comment]
> I'm not too thrilled about Flake8 homogenizing all the column-formatted code though.

I feel your pain. I used to align code too. 
That said, coding more and more Python over the years, and after having read [Guido's style guide](http://web.archive.org/web/19980526013513/http://www.python.org/doc/essays/styleguide.html) [1998, how does time fly] I came around to PEP8. You get used to it. Well, I got used to it :-). Also (Git) diffs will be less clear-cut when having lines that were changed for alignment only.
Anyway, by the time I raise the PR, you can decide.

**Comment by dkobozev** [comment]
@Blokkendoos Any luck with the update? I unexpectedly found myself with some time on my hands and thought I'd blow the dust off some of the old code. I checked out your branch and got an OpenGL context error when trying to load a file. I thought I'd check in just in case before diving further on my own.

**Comment by andyDoucette** [comment]
@dkobozev This worked for me in ubuntu 22.04.  

```sh
sudo apt install python3-wxgtk4.0 
pip3 install numpy pyopengl wxpython
python3 ./tatlin.py
```

Someone else also needed to run `sudo apt-get install build-essential libgtk-3-dev` but I didn't.

**Comment by Blokkendoos** [comment]
> @Blokkendoos Any luck with the update? I unexpectedly found myself with some time on my hands and thought I'd blow the dust off some of the old code. I checked out your branch and got an OpenGL context error when trying to load a file. I thought I'd check in just in case before diving further on my own.
Can you give details of the OpenGL error?
Here on Mac OS 10.13.6 (High Sierra) with (macports) py-opengl, only some (deprecation) warnings, see attached logs.

[2023-09-15 tatlin STL.txt](https://github.com/dkobozev/tatlin/files/12640531/2023-09-15.tatlin.STL.txt)
[2023-09-15 tatlin gcode.txt](https://github.com/dkobozev/tatlin/files/12640532/2023-09-15.tatlin.gcode.txt)

**Comment by dkobozev** [comment]
@andyDoucette, @Blokkendoos, thanks for confirming that it works on your machines, that narrowed it down significantly. A quick search revealed that setting `PYOPENGL_PLATFORM=egl` lets the context initialize and display the scene on my machine. I still need to figure out what that does exactly or why I need to set it manually on Arch in the first place.

@Blokkendoos I should be able to merge your branch in the next few days barring any surprises.

**Comment by andyDoucette** [comment]
You're very welcome.  🙂  Glad i could help.

**Comment by dkobozev** [comment]
Addressed in #24
